src/collectors/vm/prometheus_vm_msacc_collector.erl

```diff
@@ -206,69 +206,70 @@ metrics() ->
     metric(other, Data)},
-   {port_microseconds, counter,
-    "Time in microseconds spent executing ports.",
+   {port_seconds_total, counter,
+    "Total time in seconds spent executing ports.",
     metric(port, Data)},
-   {sleep_microseconds, counter,
-    "Time in microseconds spent sleeping.",
+   {sleep_seconds_total, counter,
+    "Total time in seconds spent sleeping.",
     metric(sleep, Data)},
    %% Extra states.
-   {alloc_microseconds, counter,
-    "Time in microseconds spent managing memory. "
+   {alloc_seconds_total, counter,
+    "Total time in seconds spent managing memory. "
     "Without extra states this time is spread out over all other states.",
     metric(alloc, Data)},
-   {bif_microseconds, counter,
-    "Time in microseconds spent in BIFs. "
+   {bif_seconds_total, counter,
+    "Total time in seconds spent in BIFs. "
     "Without extra states this time is part of the 'emulator' state.",
     metric(bif, Data)},
-   {busy_wait_microseconds, counter,
-    "Time in microseconds spent busy waiting. "
+   {busy_wait_seconds_total, counter,
+    "Total time in seconds spent busy waiting. "
     "Without extra states this time is part of the 'other' state.",
     metric(busy_wait, Data)},
-   {ets_microseconds, counter,
-    "Time in microseconds spent executing ETS BIFs. "
+   {ets_seconds_total, counter,
+    "Total time in seconds spent executing ETS BIFs. "
     "Without extra states this time is part of the 'emulator' state.",
     metric(ets, Data)},
-   {gc_full_microseconds, counter,
-    "Time in microseconds spent doing fullsweep garbage collection. "
+   {gc_full_seconds_total, counter,
+    "Total time in seconds spent doing fullsweep garbage collection. "
     "Without extra states this time is part of the 'gc' state.",
     metric(gc_full, Data)},
-   {nif_microseconds, counter,
-    "Time in microseconds spent in NIFs. "
+   {nif_seconds_total, counter,
+    "Total time in seconds spent in NIFs. "
     "Without extra states this time is part of the 'emulator' state.",
     metric(nif, Data)},
-   {send_microseconds, counter,
-    "Time in microseconds spent sending messages (processes only). "
+   {send_seconds_total, counter,
+    "Total time in seconds spent sending messages (processes only). "
     "Without extra states this time is part of the 'emulator' state.",
     metric(send, Data)},
-   {timers_microseconds, counter,
-    "Time in microseconds spent managing timers. "
+   {timers_seconds_total, counter,
+    "Total time in seconds spent managing timers. "
     "Without extra states this time is part of the 'other' state.",
     metric(timers, Data)}
   ].
 
 metric(Counter, Data) ->
+    SecondAsNative = erlang:convert_time_unit(1, seconds, native),
     [
         {
             [{type, Type}, {id, ID}],
-            erlang:convert_time_unit(Value, perf_counter, microsecond)
+            erlang:convert_time_unit(Value, perf_counter, native) / SecondAsNative
```

Comment:
FWIW it does appear that conversion to `native` or `nanosecond` preserves data:

```
16> erlang:convert_time_unit(os:perf_counter(), perf_counter, native).
1578692739962803000
17> erlang:convert_time_unit(os:perf_counter(), perf_counter, nanosecond).
1578692746603019500
18> erlang:convert_time_unit(os:perf_counter(), perf_counter, microsecond).
1578692752457619
```

Rather than calling `convert_time_unit/3` on a single second, conversion to `nanosecond` could be used with a hard-coded value for "second as nanosecond".

Reply:
I don't think there's a need to convert from `perf_counter` to something else. Instead we should have `SecondAsPerfCounter` and we just do `Value / SecondAsPerfCounter`. We'll only have the single `convert_time_unit` call that way.